gamemode/modules/fpp/pp/client/hud.lua

```diff
@@ -83,17 +92,17 @@ local function DrawNotice(k, v, i)
         col = Color(30, 100, 30, v.a * 0.4)
     end
 
-    draw.RoundedBox(4, x - w - h + 16, y - 8, w + h, h, col)
+    draw_RoundedBox(4, x - w - h + 16, y - 8, w + h, h, col)
 
     -- Draw Icon
 
-    surface.SetDrawColor( 255, 255, 255, v.a )
+    surface_SetDrawColor( 255, 255, 255, v.a )
 
-    draw.SimpleText(v.text, "TabLarge", x + 1, y + 1, Color(0, 0, 0, v.a * 0.8), TEXT_ALIGN_RIGHT)
-    draw.SimpleText(v.text, "TabLarge", x - 1, y - 1, Color(0, 0, 0, v.a * 0.5), TEXT_ALIGN_RIGHT)
-    draw.SimpleText(v.text, "TabLarge", x + 1, y - 1, Color(0, 0, 0, v.a * 0.6), TEXT_ALIGN_RIGHT)
-    draw.SimpleText(v.text, "TabLarge", x - 1, y + 1, Color(0, 0, 0, v.a * 0.6), TEXT_ALIGN_RIGHT)
-    draw.SimpleText(v.text, "TabLarge", x, y, Color(255, 255, 255, v.a), TEXT_ALIGN_RIGHT)
+    draw_SimpleText(v.text, "TabLarge", x + 1, y + 1, Color(0, 0, 0, v.a * 0.8), TEXT_ALIGN_RIGHT)
```

Comment:
The biggest time save in this function would be to inline all these draw.SimpleText calls. It caches internal data with the right alignment that you could only calc once for all these. Also, surface.SetDrawColor(num, num, num, num) is way more efficient than surface.SetDrawColor(Color(...)).

You can see the implementation here https://github.com/Facepunch/garrysmod/blob/34d2ccc460688750cb81c0ca031c75635aacfbaf/garrysmod/lua/includes/modules/draw.lua#L56